### PR TITLE
Refactor FXIOS-5848 [v122] Remove SnapKit from DevicePickerTableViewCell

### PR DESCRIPTION
**Pull request by bmihai23**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-5848)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/13360)

## :bulb: Description
- Remove SnapKit from DevicePickerTableViewCell and replace with normal constraints

| Portrait | Landscape | 
| ------------- | ------------- |
| portrait | landscape |

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

**Comment by bmihai23** [comment]
If you watch the attached video, you can see that I have a problem when the phone rotates to landscape mode, where the label overlaps that icon.

**Comment by PARAIPAN9** [comment]
I think the SnapKit constraints set the anchors to the content of the view by default. Therefore, the equivalent would be `contentView.leadingAnchor` and `contentView.trailingAnchor`, ensuring that the views no longer overlap.

File: Client/Frontend/DevicePickerTableViewCell.swift [hunk 1]
```diff
@@ -13,7 +13,12 @@ class DevicePickerTableViewCell: UITableViewCell, ReusableCell, ThemeApplicable
         static let deviceRowTextPaddingRight: CGFloat = 50
     }
 
-    var nameLabel: UILabel
+    var nameLabel: UILabel = .build { label in
```
Comment (PARAIPAN9) [inline Client/Frontend/DevicePickerTableViewCell.swift:16]: Please make it `private lazy var nameLabel: UILabel`

**Comment by PARAIPAN9** [comment]
Also after making the `nameLabel` var private we would need a configure method to have access to its text property, something like this: 
`func configure(_ text: String, _ clientType: ClientType) {
        nameLabel.text = text
        self.clientType = clientType
    }`

**Comment by adudenamedruby** [comment]
@PARAIPAN9 is right, however, we should maintain consistency with the rest of the codebase and named that function `configurCell(....` :D

File: Client/Frontend/DevicePickerTableViewCell.swift [hunk 2]
```diff
@@ -41,20 +42,21 @@ class DevicePickerTableViewCell: UITableViewCell, ReusableCell, ThemeApplicable
...
-        nameLabel.snp.makeConstraints { (make) -> Void in
-            make.left.equalTo(UX.deviceRowTextPaddingLeft)
-            make.centerY.equalTo(self.snp.centerY)
-            make.right.equalTo(self.snp.right).offset(-UX.deviceRowTextPaddingRight)
-        }
+        NSLayoutConstraint.activate([
+            nameLabel.leadingAnchor.constraint(equalTo: leadingAnchor, constant: UX.deviceRowTextPaddingLeft),
+            nameLabel.centerYAnchor.constraint(equalTo: centerYAnchor),
+            nameLabel.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -UX.deviceRowTextPaddingRight)
+        ])
```
Comment (PARAIPAN9) [inline Client/Frontend/DevicePickerTableViewCell.swift:49]: I also think there is room for improvement here. We don't need to activate the constraints with every layout change; instead, we could create a `setupLayout()` method, add the constraints there, and call it in the `init`. This change is arbitrary and not related to this PR, so feel free to consider it only if you want. Thank you for all the good work you put in and your constant PRs.

**Review by PARAIPAN9 — APPROVED** [review]
Great Job, LGTM. CC: @adudenamedruby

**Comment by adudenamedruby** [comment]
This looks good. There's only one improvement that I'd ask for, which is moving the `configureCell` down with the rest of the functions. We try to have separation of variables, initializers, view lifecycle, and functions. :)

**Review by adudenamedruby — APPROVED** [review]
🔥

**Comment by adudenamedruby** [comment]
Thanks so much for your contribution @bmihai23!!! :D

**Pull request merged**